.github/actions/setup/action.yml

```diff
@@ -48,6 +52,15 @@ runs:
           ${{ runner.os }}-build-${{ env.hash }}-
           ${{ runner.os }}-build-
           ${{ runner.os }}-
+    - name: Cache e2e
+      if: inputs.enable-e2e-cache == 'true'
+      uses: actions/cache@v3
+      with:
+        path: |
+          .cache/e2e
+        key: ${{ runner.os }}-e2e-${{ hashFiles('yarn.lock') }}
```

Comment:
It will be reset only in case of `yarn.lock` update.
Maybe you can add `hashFiles('**/*.it.spec.ts')` as well

Reply:
as discussed, since we don't have a good way to generate a hash key to identify the dependencies of the test-app we decided to use a self-invalidating key that will change every month